From b41e8e5afa9618e113b4c78ca9810bc979206951 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Tue, 9 Mar 2021 12:36:41 -0800
Subject: [PATCH 1770/1852] drm: xlnx: zynqmp_disp: Force alpha to 0 when
 external CRTC used

commit 53ffa3ebf0b702c49bbd8c484ce34c4b04685c5f from
https://github.com/Xilinx/linux-xlnx.git

If PS DP CRTC is not used, the global alpha will default to 255, thus
blending out the video layer. To avoid this, set the global alpha to 0
when an external CRTC is connected using the video layer.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Signed-off-by: Jianqiang Chen <jianqiang.chen@xilinx.com>
Acked-by: Varunkumar Allagadapa <varunkumar.allagadapa@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_disp.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_disp.c b/drivers/gpu/drm/xlnx/zynqmp_disp.c
index 652335bd8745..80cf40cca5c0 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_disp.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_disp.c
@@ -3043,6 +3043,11 @@ static int zynqmp_disp_bridge_enable(struct xlnx_bridge *bridge)
 	if (disp->dpsub->external_crtc_attached)
 		zynqmp_disp_crtc_atomic_enable(crtc, NULL);
 
+	/* If external CRTC is connected through video layer, set alpha to 0 */
+	if (disp->dpsub->external_crtc_attached &&
+		layer->id == ZYNQMP_DISP_LAYER_VID)
+		disp->alpha = 0;
+
 	zynqmp_disp_set_g_alpha(disp, disp->alpha_en);
 	zynqmp_disp_set_alpha(disp, disp->alpha);
 	ret = zynqmp_disp_layer_enable(layer->disp, layer,
-- 
2.31.1

